packages/playground/src/components/account/account-register/account-register.tsx

```diff
@@ -78,16 +79,21 @@ export class AccountRegister {
     this.changeset[key] = (event.target as HTMLInputElement).value;
   }
 
-  formSubmissionHandler() {
+  async formSubmissionHandler() {
     const data = this.changeset;
 
-    // We use personal#sign() because eth#sign() is dangerous.
-    // See: https://metamask.zendesk.com/hc/en-us/articles/360015488751
-    window["web3"].personal.sign(
-      window["web3"].toHex(buildRegistrationSignaturePayload(data)),
-      data.ethAddress,
-      this.register.bind(this)
-    );
+    const payload = buildRegistrationSignaturePayload(data);
+
+    this.metamaskConfirmationUIOpen = true;
+
+    try {
+      const signature = await this.signer.signMessage(payload);
+      await this.register(signature);
+    } catch (e) {
+      await this.register("", e);
```

Comment:
@snario What's the purpose of this? Seems a bit counter-intuitive.

Reply:
The way `register` was written is that is gets called with two arguments; an error or a signed message and it handles it differently depending. Pretty easy to refactor. I'll refactor.